src/jit/inlinepolicy.cpp

```diff
@@ -272,3 +306,75 @@ void LegacyPolicy::setNever(InlineObservation obs)
         unreached();
     }
 }
+
+//------------------------------------------------------------------------
+// determineMultiplier: determine benefit multiplier for this inline
+//
+// Notes: uses the accumulated set of observations to compute a
+// profitability boost for the inline candidate.
+
+double LegacyPolicy::determineMultiplier()
+{
+    double multiplier = 0;
+
+    // Bump up the multiplier for instance constructors
+
+    if (inlIsInstanceCtor)
+    {
+        multiplier += 1.5;
+        JITDUMP("\nmultiplier in instance constructors increased to %g.", multiplier);
+    }
+
+    // Bump up the multiplier for methods in promotable struct
+
+    if (inlIsFromPromotableValueClass)
+    {
+        multiplier += 3;
+        JITDUMP("\nmultiplier in methods of promotable struct increased to %g.", multiplier);
+    }
+
+#ifdef FEATURE_SIMD
```

Comment:
I'd like to have as few ifdefs as possible, so they are over at the observation site but not elsewhere.